### PR TITLE
Update CHANGELOG for new generics features in Swift 5.7

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,86 @@ _**Note:** This is in reverse chronological order, so newer entries are added to
...
+
+  ```swift
+    protocol Graph<Vertex, Edge> {
+      associatedtype Vertex
+      associatedtype Edge
+    }
+  ```
+
+  A protocol-constrained type like `Graph<Int>` can now be written anywhere that
+  expects the right-hand side of a protocol conformance requirement:
```
Comment (AnthonyLatsis) [inline CHANGELOG.md:48]: I find the other half of this sentence a bit hard to parse through. I think it could take developers more than expected to make sense of it. What is your take on rewording it to one of the following?

* Similar to a protocol type, a protocol-constrained type like `Graph<Int>` can be extended or used as a protocol conformance requirement
* Similar to a protocol type, a protocol-constrained type like `Graph<Int>` can be extended, used with `some` or at the right-hand side of a protocol conformance requirement
